### PR TITLE
Issue37 multigpu

### PR DESCRIPTION
**Pull request by lissyx**
issue #37

### CODE REVIEW

**Comment by kdavis-mozilla** [comment]
Should I review this as if you want it merged into master, or is this more a work in progress?

**Comment by lissyx** [comment]
Still WIP I will rebase on master once multigpu batch_size support is merged, and then I will be able to run and update the results with multi gpu :)

**Comment by kdavis-mozilla** [comment]
If it's just a WIP, maybe we shouldn't have it as a pull request?

**Comment by lissyx** [comment]
Why closing it ?

**Comment by kdavis-mozilla** [comment]
It's not really a pull request. The issue it sights is issue #37 "Benchmark the Warp-ctc algorithm"

This pull requests does much more than that. It regresses changes already in master, such a multi-gpu support, requires that all future code benchmarks, introduces a new parallel neural network `Network` is incompatible with our current neural network,  introduces `deepspeech.py` which doesn't support multiple GPU'S....

Basically there are a lot of change here and very few of them have anything do do with benchmarking. In addition most of the changes will break other pull requests that are working within the parameters of master framework.

**Comment by lissyx** [comment]
I am in the process of working on this. My PR is waiting for your PR #50 to land so that I can rebase on this, and effectively get the Multi-GPU support. Yes, there is old code that is incompatible, this is what I used to start working on benchmark since not everything was ready.

**Comment by kdavis-mozilla** [comment]
Multi GPU support is already in master and your code is removing it, see `get_tower_gradients()`
